lib/ansible/modules/network/asa/asa_og.py

```diff
@@ -0,0 +1,804 @@
+        type: list
+    description:
+        description:
+            - object-group description
+    group_object:
+        description:
+            - group-object for network object-group i.e. "network-object object-group ansible_test"
+        type: list
+    ip_mask:
+        description:
+            - IP address and mask for network object-group i.e. "network-object 192.168.0.0 255.255.0.0"
+        type: list
+    port_range:
+        description:
+            - port range for port-object i.e. "port-object range 0 1024"
+    port_eq:
+        description:
+            - single port for port-object i.e. "port-object eq www"
+    service_cfg:
+        description:
+            - service-object configuration protocol/direction/range,port/port
+    state:
+        description:
+            - Manage the state of the resource.
+        default: replace
+        choices: ['present', 'absent', 'replace']
+"""
+
+EXAMPLES = """
+# Note: examples below use the following provider dict to handle
+#       transport and authentication to the node.
+---
+- name: configure network object-group
+  asa_og:
+    name: ansible_test_0
+    group_type: network-object
+    state: present
+    description: ansible_test object-group description
+    host_ip:
+      - 8.8.8.8
+      - 8.8.4.4
+    ip_mask:
+      - 10.0.0.0 255.255.255.0
+      - 192.168.0.0 255.255.0.0
+    group_object:
+      - awx_lon
+      - awx_ams
+  register: result
+
+- name: configure port-object object-group
+  asa_og:
+    name: ansible_test_1
+    group_type: port-object
+    state: replace
+    description: ansible_test object-group description
+    protocol: tcp-udp
+    port_eq:
+      - 1025
+      - kerberos
+    port_range:
+      - 1025 5201
+      - 0 1024
+  register: result
+
+- name: configure service-object object-group
+  asa_og:
+    name: ansible_test_2
+    group_type: service-object
+    state: absent
+    description: ansible_test object-group description
+    service_cfg:
+      - tcp destination eq 8080
+      - tcp destination eq www
+  register: result
+"""
+
+RETURN = """
+commands:
+  description: command sent to the device
+  returned: always
+  type: list
+  sample: [
+    "object-group network ansible_test_0",
+    "description ansible_test object-group description",
+    "network-object host 8.8.8.8",
+    "network-object host 8.8.4.4",
+    "network-object 10.0.0.0 255.255.255.0",
+    "network-object 192.168.0.0 255.255.0.0",
+    "network-object 192.168.0.0 255.255.0.0",
+    "group-object awx_lon",
+    "group-object awx_ams",
+    ]
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.asa.asa import asa_argument_spec, check_args
+from ansible.module_utils.network.asa.asa import get_config, load_config, run_commands
+from ansible.module_utils.network.common.config import NetworkConfig, dumps
+import re
+import sys
+__metaclass__ = type
```

Comment:
Please move this below the future import statement, to keep the module consistent.

Reply:
All the imports must og under __future__ or just those indicated in your comment?

Reply:
@NilashishC 
If I change the import order, pipeline fails

```
lib/ansible/modules/network/asa/asa_og.py:8:0: E106 Import found before documentation variables. All imports must appear below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA.
lib/ansible/modules/network/asa/asa_og.py:8:0: E107 Imports should be directly below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA.
```

Reply:
@FedericoOlivieri I meant something like [this](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/asa/asa_config.py#L6).

Reply:
gotcha!